lib/eslint.js

```diff
@@ -407,6 +407,30 @@ function isDisabledByReportingConfig(reportingConfig, ruleId, location) {
 }
 
 /**
+ * Normalize ECMAScript version from the initial config
+ * @param  {number} ecmaVersion ECMAScript version from the initial config
+ * @param  {boolean} isModule Whether the source type is module or not
+ * @returns {number} normalized ECMAScript version
+ */
+function normalizeEcmaVersion(ecmaVersion, isModule) {
+
+    // need at least ES6 for modules
+    if (isModule && (!ecmaVersion || ecmaVersion < 6)) {
+        ecmaVersion = 6;
+    }
+
+    if (ecmaVersion === 2016) {
```

Comment:
I'd just say `if (ecmaVersion >= 2015) ecmaVersion -= 2009`. That way we don't have to keep updating this section.